frontend/src/metabase/admin/tasks/components/LogLevelsModal/DurationInput.tsx

```diff
@@ -0,0 +1,61 @@
+import { t } from "ttag";
+
+import { Flex, Select, TextInput } from "metabase/ui";
+import type { TimeUnit } from "metabase-types/api";
+
+interface Props {
+  duration: string;
+  durationUnit: TimeUnit;
+  onDurationChange: (duration: string) => void;
+  onDurationUnitChange: (durationUnit: TimeUnit) => void;
+}
+
+export const DurationInput = ({
+  duration,
+  durationUnit,
+  onDurationChange,
+  onDurationUnitChange,
+}: Props) => {
+  return (
+    <Flex align="flex-end" gap="md">
+      <TextInput
+        label={t`Duration`}
+        placeholder={t`Duration`}
+        required
+        type="number"
+        value={duration}
+        w={80}
+        onChange={(event) => onDurationChange(event.target.value)}
+      />
+
+      <Select
+        data={getData()}
+        placeholder={t`Unit`}
+        value={durationUnit}
+        w={140}
+        onChange={onDurationUnitChange}
+      />
+    </Flex>
+  );
+};
+
+// We don't want to show nanoseconds in the dropdown
+type AllowedTimeUnit = Exclude<TimeUnit, "nanoseconds">;
+
+// It intentionally is a function and not a module-level constant, so that `t` function works correctly
+function getData() {
+  /**
+   * Using a Record, so that this gives compilation error when TimeUnit is extended,
+   * so that whoever changes that type does not forget to update this function.
+   */
+  const statusNames: { [T in AllowedTimeUnit]: { label: string; value: T } } = {
+    microseconds: { label: t`Microseconds`, value: "microseconds" },
```

Comment:
Same as above, I'd also remove micro and milliseconds unless someone things anything can be captured within such a small time period

Reply:
4ba70d12851d74f34b4e2384b5942fa22b45bc57